src/test/kotlin/uk/gov/dluhc/printapi/database/repository/CertificateRepositoryIntegrationTest.kt

```diff
@@ -179,6 +181,44 @@ internal class CertificateRepositoryIntegrationTest : IntegrationTest() {
         assertCertificateRecursiveEqual(actual[0], expected)
     }
 
+    @Nested
+    inner class FindByGssCodeInAndSourceTypeAndSourceReference {
+        @Test
+        fun `should find certificate given one exists for provided details`() {
```

Comment:
Not part of this ticket, I assume, but is there a unique index in the DB to prevent multiple certificates with the same source type and source reference and gssCode? (I guess we cannot have two certificates with the same source type and sourceReference alone). I think it will be good to prevent that in the db layer.